packages/flutter/lib/src/widgets/reorderable_list.dart

```diff
@@ -730,6 +730,9 @@ class SliverReorderableListState extends State<SliverReorderableList> with Ticke
   }
 
   void _registerItem(_ReorderableItemState item) {
+    if (_dragInfo != null && _items[item.index] == null) {
```

Comment:
what if _items[item.index] contains other item? e.g a rebuild that replace an existing item with a new item at a index

Reply:
You are right, update to `_items[item.index] != item`